### PR TITLE
Add suggestive MLA and APA citations for records

### PR DESCRIPTION
**Pull request by edejesusyale**
**STORY**
As an undergraduate writing a paper, I would like the system to provide properly formatted citations, so that I'm less likely to make errors transcribing or formatting my citations.

**ACCEPTANCE**
Go to any record's display/item page.
Click on the "Cite" link in the Tools box on a single item show view and verify:
- [ ] You get a pop-up window with the citations for the work
  - [ ] Citation in MLA (Modern Language Association) format
  - [ ] Citation in APA (American Phycological Association) format

### CODE REVIEW

**Comment by alishaevn** [comment]
approving, but not merging since I paired on this with Eric.

File: lib/yale/citation_formatter.rb [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+
+    def initialize(obj)
+      @obj = obj
+      @default_citations = {
+        "apa": apa_default_citation,
+        "modern-language-association": mla_default_citation
+      }
+    end
+
+    def citation_for(style)
```
Comment (bess) [inline lib/yale/citation_formatter.rb:20]: Please document what the acceptable values of `style` are here

File: lib/yale/citation_formatter.rb [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+require 'citeproc'
+require 'csl/styles'
+require './lib/yale/citation_string_processor.rb'
+
+module Yale
+  class CitationFormatter
+    include CitationStringProcessor
+    attr_accessor :obj, :default_citations
+
+    def initialize(obj)
```
Comment (bess) [inline lib/yale/citation_formatter.rb:12]: Please document what we expect `obj` to be here

File: lib/yale/citation_formatter.rb (hunk 1 above)
Comment (alishaevn) [reply lib/yale/citation_formatter.rb]: @bess just calling your attention back to this. @edejesusyale followed through on your 2 comments above. 🙂

File: lib/yale/citation_formatter.rb (hunk 2 above)
Comment (alishaevn) [reply lib/yale/citation_formatter.rb]: @bess just calling your attention back to this. @edejesusyale followed through on your 2 comments above. 🙂

**Pull request merged**